### ISSUE TITLE
Batch normalization layer have 4 parameters. 

### ISSUE THREAD

**Issue by hafssol** [fchollet/keras#1523]
When I execute model.get_weights() , It seems each BN layer have 4 parameters serially 
I believe there are gamma , inverse stdev ,  Gamma , beta are included. 
But I cannot sure the order of the parameter. 
what is the first parameter?
what is the second parameter?
what is the third parameter?
what is the fourth parameter?
It will be grateful for me to know.

**Comment by fchollet** [comment]
> But I cannot sure the order of the parameter. 

Well, read the source code! It's straightforward.

**Comment by hafssol** [comment]
well I read it hours. But I cannot understood. (i'm not good at code.)
please give me a hint.

**Comment by AvantiShri** [comment]
Looks like gamma, beta, running mean and running std is the order (super.get_weights() returns self.params and then running mean and std are tacked on): https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L61-L72

**Comment by bluebirdlboro** [comment]
AvantiShri is right. And you can find the answer from this keras document
https://faroit.github.io/keras-docs/1.2.2/layers/normalization/
It explains weights order is : [gamma, beta, mean, std]
Somehow, another keras document https://keras.io/layers/normalization/ lacks of the note for the 'weights' parameter.

**Comment by dddhiraj** [comment]
Looking at code at https://github.com/keras-team/keras/blob/master/keras/layers/normalization.py#L61-L72

I think it should be [gamma, beta, mean , variance]

**Comment by rsnayaksd** [comment]
> When I execute model.get_weights() , It seems each BN layer have 4 parameters serially
> I believe there are gamma , inverse stdev , Gamma , beta are included.
> But I cannot sure the order of the parameter.
> what is the first parameter?
> what is the second parameter?
> what is the third parameter?
> what is the fourth parameter?
> It will be grateful for me to know.

**Comment by rsnayaksd** [comment]
In case of Batch Normalization get_weights() will provide four additional parameter values for each hidden layers in neural network. 

Please refer link [https://stackoverflow.com/questions/57087273/can-you-explain-keras-get-weights-function-in-a-neural-network-with-batchnorma?answertab=active#tab-top](url) for more details.

**Comment by jennakwon06** [comment]
> > But I cannot sure the order of the parameter.
> 
> Well, read the source code! It's straightforward.

"Straight forward" is subjective & "Read the source code" is not a great reply. Well written and clear code documentation is expected for any usable codebase and what you are saying is that it's OK for many people to be wasting their hours sifting through the codebase..

**Comment by Yanagar1** [comment]
the last one is variance!
